hudi-common/src/main/java/org/apache/hudi/internal/schema/InternalSchemaBuilder.java

```diff
@@ -67,6 +68,10 @@ public Map<String, Integer> buildNameToId(Type type) {
     return visit(type, new NameToIDVisitor());
   }
 
+  Map<String, Integer> buildNameToPosition(Type type) {
```

Comment:
High level question: Do we use the InternalSchemaBuilder even when schema on read is disabled?

Reply:
The internal schema seems to provide some nice utilities but I was not familiar with it before this change

Reply:
We use internal schema even when schema on read is disabled. We use it to add null for missing columns, promote incoming batch if it can be promoted to the table schema, and also to fix the ordering of unions